test_*_on_large_data_send errors out on 32-bit platforms

When running the test suite on 32-bit x86, Python errors out due to overflow:

```
=================================== FAILURES ===================================
_ TestsProcessPoolForkserverExecutor.test_expected_failure_on_large_data_send __

self = <tests.test_process_executor_forkserver.TestsProcessPoolForkserverExecutor object at 0xf394fbec>

    @pytest.mark.high_memory
    @pytest.mark.skipif(
            sys.version_info[:2] >= (3, 8),
            reason="These Pythons can pickle objects of size > 2 ** 31GB")
    def test_expected_failure_on_large_data_send(self):
>       data = b'\x00' * int(2.2e9)
E       OverflowError: cannot fit 'int' into an index-sized integer
```

I suppose the test should be skipped on 32-bit platforms.

Full log: https://743334.bugs.gentoo.org/attachment.cgi?id=661152

--- a/tests/_test_process_executor.py
+++ b/tests/_test_process_executor.py
@@ -903,17 +903,21 @@ class ExecutorTest:
                                  ctx=self.context)
 
     @pytest.mark.high_memory
+    @pytest.mark.skipif(sys.maxsize < 2 ** 32,
+                        reason="Test requires a 64 bit version of Python")
     @pytest.mark.skipif(
             sys.version_info[:2] < (3, 8),
-            reason="These Pythons cannot pickle objects of size > 2 ** 31GB")
+            reason="Python version does not support pickling objects of size > 2 ** 31GB")
     def test_no_failure_on_large_data_send(self):
         data = b'\x00' * int(2.2e9)
         self.executor.submit(id, data).result()
 
     @pytest.mark.high_memory
+    @pytest.mark.skipif(sys.maxsize < 2 ** 32,
+                        reason="Test requires a 64 bit version of Python")
     @pytest.mark.skipif(
             sys.version_info[:2] >= (3, 8),
-            reason="These Pythons can pickle objects of size > 2 ** 31GB")
+            reason="Python version supports pickling objects of size > 2 ** 31GB")
     def test_expected_failure_on_large_data_send(self):
         data = b'\x00' * int(2.2e9)
         with pytest.raises(RuntimeError):

